src/lipizzaner_master.py

```diff
@@ -157,85 +162,112 @@ def _terminate(self, stop_clients=True, return_code=-1):
             exit(return_code)
 
     def _gather_results(self):
-        self._logger.info('Collecting results from clients...')
+        self._logger.info("Collecting results from clients...")
 
         # Initialize node client
-        dataloader = self.cc.create_instance(self.cc.settings['dataloader']['dataset_name'])
-        network_factory = self.cc.create_instance(self.cc.settings['network']['name'], dataloader.n_input_neurons)
+        dataloader = self.cc.create_instance(self.cc.settings["dataloader"]["dataset_name"])
+        network_factory = self.cc.create_instance(
+            self.cc.settings["network"]["name"], dataloader.n_input_neurons, num_classes=dataloader.num_classes,
+        )
         node_client = NodeClient(network_factory)
         db_logger = DbLogger()
 
-        results = node_client.gather_results(self.cc.settings['general']['distribution']['client_nodes'], 120)
+        results = node_client.gather_results(self.cc.settings["general"]["distribution"]["client_nodes"], 120)
 
         scores = []
-        for (node, generator_pop, discriminator_pop, weights_generator, weights_discriminator) in results:
-            node_name = '{}:{}'.format(node['address'], node['port'])
+        for (node, generator_pop, discriminator_pop, weights_generator, weights_discriminator,) in results:
+            node_name = "{}:{}".format(node["address"], node["port"])
             try:
                 output_dir = self.get_and_create_output_dir(node)
 
                 for generator in generator_pop.individuals:
-                    source = generator.source.replace(':', '-')
-                    filename = '{}{}.pkl'.format(GENERATOR_PREFIX, source)
-                    torch.save(generator.genome.net.state_dict(),
-                               os.path.join(output_dir, 'generator-{}.pkl'.format(source)))
+                    source = generator.source.replace(":", "-")
+                    filename = "{}{}.pkl".format(GENERATOR_PREFIX, source)
+                    torch.save(
+                        generator.genome.net.state_dict(), os.path.join(output_dir, "generator-{}.pkl".format(source)),
+                    )
 
-                    with open(os.path.join(output_dir, 'mixture.yml'), "a") as file:
-                        file.write('{}: {}\n'.format(filename, weights_generator[generator.source]))
+                    with open(os.path.join(output_dir, "mixture.yml"), "a") as file:
+                        file.write("{}: {}\n".format(filename, weights_generator[generator.source]))
 
                 for discriminator in discriminator_pop.individuals:
-                    source = discriminator.source.replace(':', '-')
-                    filename = '{}{}.pkl'.format(DISCRIMINATOR_PREFIX, source)
-                    torch.save(discriminator.genome.net.state_dict(),
-                               os.path.join(output_dir, filename))
+                    source = discriminator.source.replace(":", "-")
```

Comment:
Should this be made a method for the network to call `save`?

Reply:
We did not change that, but I created the function.